Storage/src/Lifecycle.php

```diff
@@ -0,0 +1,236 @@
+     *     @type int $numNewerVersions Relevant only for versioned objects. If
+     *           the value is N, this condition is satisfied when there are at
+     *           least N versions (including the live version) newer than this
+     *           version of the object.
+     * }
+     * @return Lifecycle
+     */
+    public function addSetStorageClassRule($storageClass, array $condition)
+    {
+        $this->lifecycle['rule'][] = [
+            'action' => [
+                'type' => 'SetStorageClass',
+                'storageClass' => $storageClass
+            ],
+            'condition' => $condition
+        ];
+
+        return $this;
+    }
+
+    /**
+     * Clears out rules based on the provided condition.
+     *
+     * Example:
+     * ```
+     * // Remove all rules.
+     * $lifecycle->clearRules();
+     * ```
+     *
+     * ```
+     * // Remove all "Delete" based rules.
+     * $lifecycle->clearRules('Delete');
+     * ```
+     *
+     * @param string|callable $condition [optional] If a string is provided, it
+     *        must be the name of the type of rule to remove (`SetStorageClass`
+     *        or `Delete`). All rules of this type will then be cleared. When
+     *        providing a callable you may define a custom route for how you
+     *        would like to remove rules. The provided callable will be run
+     *        through
+     *        [array_filter](http://php.net/manual/en/function.array-filter.php).
+     *        The callable's argument will be a single lifecycle rule as an
+     *        associative array. When returning true from the callable the rule
+     *        will be preserved, and if false it will be removed.
+     *        **Defaults to** `null`, clearing all assigned rules.
+     * @return Lifecycle
+     * @throws \InvalidArgumentException If a type other than a string or
+     *         callabe is provided.
+     */
+    public function clearRules($condition = null)
+    {
+        if (!$condition) {
+            $this->lifecycle = [];
+            return $this;
+        }
+
+        if (!is_string($condition) && !is_callable($condition)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Expected either a string or callable, instead got \'%s\'.',
+                    gettype($condition)
+                )
+            );
+        }
+
+        if (isset($this->lifecycle['rule'])) {
+            if (is_string($condition)) {
+                $condition = function ($rule) use ($condition) {
+                    return $rule['action']['type'] !== $condition;
+                };
+            }
+
+            $this->lifecycle['rule'] = array_filter(
+                $this->lifecycle['rule'],
+                $condition
+            );
+
+            if (!$this->lifecycle['rule']) {
+                $this->lifecycle = [];
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * @access private
```

Comment:
Is the following toArray() private? Not sure what this comment section is referring to.

Reply:
It is marked this way to let users know it is intended for internal use only.